mlrun/feature_store/api.py

```diff
@@ -528,6 +530,8 @@ def ingest(
         raise mlrun.errors.MLRunInvalidArgumentError(
             "featureset.spec.engine must be set to 'spark' to ingest with spark"
         )
+    if featureset is not None:
```

Comment:
Note that this location is very late in the function - I thought about doing this verification much earlier, right after the `featureset` is extracted in `context_to_ingestion_params` (also I think at this point `featureset` must exist, so no need to do the `is not None` check).

Reply:
done! tnx Saar